controllers/storagecluster/generate.go

```diff
@@ -79,7 +80,22 @@ func generateNameForSnapshotClassDriver(initData *ocsv1.StorageCluster, snapshot
 	return fmt.Sprintf("%s.%s.csi.ceph.com", initData.Namespace, snapshotType)
 }
 
-func generateNameForSnapshotClassSecret(snapshotType SnapshotterType) string {
+func generateNameForSnapshotClassSecret(instance *ocsv1.StorageCluster, snapshotType SnapshotterType) string {
+	if instance.Spec.ExternalStorage.Enable == true {
+		data, ok := externalOCSResources[instance.UID]
+		if !ok {
+			log.Error(fmt.Errorf("Unable to retrieve external resource from externalOCSResources"),
+				"unable to generate name for snapshot class secret for external mode")
+		}
+		// print the Secret name which contains the prefix as the rook-csi-rbd/cephfs-provisioner default secret name
+		for _, d := range data {
+			if d.Kind == "Secret" {
+				if strings.Contains(d.Name, fmt.Sprintf("rook-csi-%s-provisioner", snapshotType)) {
+					return d.Name
```

Comment:
How does `d.Name` compare to the default name? Maybe add an example to the comment.